analyzer/linux/lib/common/results.py

```diff
@@ -44,13 +44,41 @@ def upload_to_host(file_path, dump_path, pids="", ppids="", metadata="", categor
             nc.close()
 
 
+def append_buffer_to_host(buffer, nc=None):
+    if nc.sock is None:
+        raise ConnectionResetError
+
+    size = len(buffer)
+    if config.upload_max_size < size + nc.buffer_size and not config.do_upload_max_size:
+        log.warning("Buffer is too big: %d; max size: %d", size + nc.buffer_size, config.upload_max_size)
+        return
+
+    try:
+        idx = 0
+        while idx < size:
+            nc.send(buffer[idx : idx + BUFSIZE], retry=True)
+            idx += BUFSIZE
+    except Exception as e:
```

Comment:
Might be some room for improvement here - couple suggestions:

- Move the try/except around `nc.send()` where you're expecting exceptions
- If `nc.send()` does fail, I doubt `retry=True` is helpful and only a piece of the buffer isn't too useful. Maybe make this an error and return from the function?